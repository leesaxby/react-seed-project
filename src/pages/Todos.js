import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { fetchTodos } from 'Modules/todos';
import { Grid } from '@material-ui/core';
import HeaderRow from './todos/HeaderRow';
import InputRow from './todos/InputRow';
import ListRow from './todos/ListRow';


export class Todos extends React.Component {
    componentDidMount() {
        const { fetchData } = this.props;
        fetchData();
    }

    render() {
        return (
            <div>
                <Grid container spacing={0} justify="center">
                    <Grid item xs={12} md={4}>
                        <HeaderRow />
                    </Grid>
                </Grid>
                <Grid container spacing={0} justify="center">
                    <Grid item xs={12} md={4} lg={2}>
                        <InputRow />
                    </Grid>
                </Grid>
                <Grid container spacing={0} justify="center">
                    <Grid item xs={12} md={4} lg={2}>
                        <ListRow />
                    </Grid>
                </Grid>
            </div>
        );
    }
}

Todos.propTypes = {
    fetchData: PropTypes.func.isRequired,
};

const mapDispatchToProps = (dispatch) => ({
    fetchData: () => dispatch(fetchTodos()),
});

export default connect(undefined, mapDispatchToProps)(Todos);
