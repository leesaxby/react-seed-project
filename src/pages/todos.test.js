import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mountWithIntl } from '../../test/intl-enzyme-test-uitl';
import { renderedInWrapper } from '../../test/utils';

import { Todos } from './Todos';
import TodoFilter from './todos/Filter';
import TodoList from './todos/List';
import TodoItem from './todos/Item';
import TodoForm from './todos/Form';

configure({ adapter: new Adapter() });

const props = {
    listItems: [
        { _id: 1, text: 'Item 1', done: true }
    ],
    filter: 'ACTIVE',
    updateFilter: jest.fn(),
    fetchData: jest.fn(),
    addTodo: jest.fn(),
    updateDoneStatus: jest.fn(),
    history: {}
};

const wrapper = mountWithIntl(
    <Todos { ...props }/>
  );

const isRendered = renderedInWrapper(wrapper);

describe('Todos component', () => {

    describe('Child components render', () => {

        it('Should render TodoForm component', () => {
            expect( isRendered(TodoForm) ).toBe(true);
        });

        it('Should render TodoFilter component', () => {
            expect( isRendered(TodoFilter) ).toBe(true);
        });

        it('Should render TodoList component', () => {
            expect( isRendered(TodoList) ).toBe(true);
        });

        it('Should render TodoList component', () => {
            expect( isRendered(TodoItem) ).toBe(true);
        });

    });

    describe('Calls correct props', () => {

        it('Should call addTodo prop when addTodoItem method is called', () => {
            expect(props.addTodo.mock.calls.length).toBe(0);

            wrapper.instance().addTodoItem();

            expect( props.addTodo.mock.calls.length ).toBe(1);
        });

        it('Should call fetchData when component mounts', () => {
            expect( props.fetchData.mock.calls.length ).toBe(1);
        });

    });

});
