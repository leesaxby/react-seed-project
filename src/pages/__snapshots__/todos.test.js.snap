// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A Todos Container renders component 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-1 col-lg-offset-5 col-sm-1 col-sm-offset-4"
      >
        <img
          alt="List Image"
          className=""
          src="test-file-stub"
          style={
            Object {
              "height": "200px",
              "marginBottom": "20px",
            }
          }
        />
      </div>
    </div>
    <InputRow />
    <div
      className="row"
    >
      <div
        className="col-lg-6 col-lg-offset-3 col-sm-8 col-sm-offset-2"
      >
        <List />
      </div>
    </div>
  </div>
</div>
`;

exports[`A Todos renders component 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-lg-1 col-lg-offset-5 col-sm-1 col-sm-offset-4"
      >
        <img
          alt="List Image"
          className=""
          src="test-file-stub"
          style={
            Object {
              "height": "200px",
              "marginBottom": "20px",
            }
          }
        />
      </div>
    </div>
    <InputRow />
    <div
      className="row"
    >
      <div
        className="col-lg-6 col-lg-offset-3 col-sm-8 col-sm-offset-2"
      >
        <List />
      </div>
    </div>
  </div>
</div>
`;
